norcams/himlar
# Ceph - upgrade from jewel to luminous release

These changes will upgrade the ceph cluster to the luminous release - 12.2.0

To test:
- disable puppet on cephmon nodes and storage nodes
- deploy code _AND_ secrets to admin node
- delete the puppet-ceph module: rm -rf /etc/puppet/modules/ceph
- deploy new puppet modules to admin

*** NOTE: Alle cephmons _must_ be upgraded before any storage nodes!
*** IMPORTANT: Before upgrade, run the following command on any cephmon node:

ceph osd set sortbitwise

- run upgrade ansible job for cephmons:
sudo ansible-playbook -e "hosts={loc}-cephmon" lib/upgrade/cephmon.yaml
- check for status after all cephmon nodes have been upgraded and rebooted
- then, run the ansible job for upgrading storage nodes:
sudo ansible-playbook -e "hosts={loc}-storage" lib/upgrade/storage.yaml

After all jobs have finished, the cluster will have a health warning. On any cephmon node, run:

ceph osd require-osd-release luminous

Now, all cephmons and storage nodes have been updated and rebooted. The ceph clients don't need to upgraded.
